druid/examples/image.rs

```diff
@@ -12,44 +12,319 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! This example shows how to draw an png image.
+//! This example shows how to draw an image; using images requires
+//! enabling the `image` feature in your Cargo.toml; you will also need
+//! to specify the image formats you would like to use. See druid/Cargo.toml
+//! to see all features.
 
-use druid::{
-    widget::{FillStrat, Flex, Image, WidgetExt},
-    AppLauncher, Color, ImageBuf, Widget, WindowDesc,
-};
+/* use druid::piet::InterpolationMode;
+use druid::widget::prelude::*;
+use druid::widget::{FillStrat, Flex, Image, WidgetExt};
+use druid::{AppLauncher, Color, ImageBuf, WindowDesc};
 
 pub fn main() {
     let main_window = WindowDesc::new(ui_builder);
-    let data = 0_u32;
     AppLauncher::with_window(main_window)
         .use_simple_logger()
-        .launch(data)
+        .launch(0)
         .expect("launch failed");
 }
 
 fn ui_builder() -> impl Widget<u32> {
     let png_data = ImageBuf::from_data(include_bytes!("./assets/PicWithAlpha.png")).unwrap();
 
-    let mut col = Flex::column();
-
-    col.add_flex_child(
-        Image::new(png_data.clone())
-            .border(Color::WHITE, 1.0)
-            .fix_width(100.0)
-            .center(),
-        1.0,
-    );
-
-    /*
-    // If you want to change the fill stratagy you can but you need the widget to be mut
-    let mut otherimage = Image::new(png_data);
-    otherimage.set_fill(FillStrat::FitWidth);
-    */
-
-    let otherimage = Image::new(png_data)
-        .fill_mode(FillStrat::FitWidth)
-        .border(Color::WHITE, 1.0);
-    col.add_flex_child(otherimage, 1.0);
-    col
+    // We create two images, one not having any modifications and the other
+    // forced to a fixed width, a fill strategy and an interpolation mode.
+    // You can see how this affects the final result. You can play with the
+    // interpolation mode to see how this affects things. (Note that this image
+    // is already anti-aliased so NearestNeighbor looks weird.)
+    Flex::column()
+        .with_flex_child(
+            Image::new(png_data.clone())
+                .fill_mode(FillStrat::FitWidth)
+                .interpolation_mode(InterpolationMode::NearestNeighbor)
+                .border(Color::WHITE, 1.0)
+                .fix_width(150.0)
+                .center(),
+            1.0,
+        )
+        .with_flex_child(
+            Image::new(png_data)
+                .border(Color::WHITE, 1.0),
+            1.0,
+        )
+}
+ */
+
+use druid::piet::InterpolationMode;
+use druid::widget::{prelude::*, FillStrat, Image};
+use druid::widget::{
+    Checkbox, CrossAxisAlignment, Flex, Label, RadioGroup, SizedBox, Stepper, TextBox, WidgetExt,
+};
+use druid::{AppLauncher, Color, Data, ImageBuf, Lens, LensExt, WindowDesc};
+
+const FILL_STRAT_OPTIONS: [(&str, FillStrat); 7] = [
```

Comment:
instead of having these be `const` arrays, I would have them be `static` slices; it takes up less space in the generated binary, and it doesn't require you to specify the length.

```suggestion
static FILL_STRAT_OPTIONS: &[(&str, FillStrat)] = &[
```

Reply:
does it really take up less space? but yes that sounds like a good idea, I didn't like the fixes lengths.